docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,194 @@
+
+The radical promise frequently failed on two fronts: understanding users, understanding data.
+
+### 2.2 Know your users
+
+It may seem obvious that knowing what your users are doing with your systems is a necessary prerequisite to designing an information retrieval system. But until recently, in the context of RAG at least, this hasn't been discussed much.
+
+In general, RAG applications underestimate the inertia of user practices; users continue behaving the same way they did with older expert systems. I recently analyzed a set of internal user queries in a production chatbot, and was surpised to see that many users were not submitting questions but unstructured keywords. This practice works perfectly in a standard information retrieval system, but instruction-based LLM and embedding models are designed to be trained on question/answer pairs.
+
+Inattentiion to user - design friction in LLMs and embedding models can result in poor performance, disappointed user expectations, tokenization that fails to adequately represent data structure, incongruity of input vs output lengths, and an overall lack of control over the system's purpose (the infamous “Chevrolet” effect).
+
+### 2.3 Know your data
+
+Like your users, the data will not necessarily agree with the constraints of language models and embedding models.
+
+Context window is an obvious restriction. Despite its gradual lengthening, the indexation of long texts always underperforms. This is one of the biggest mismatches between pre-training data and data in production: the vast majority of Common Crawl texts are less than 512 tokens. Organizations routinely deal with multi-page PDFs.
+
+So you need to split or chunk your document, which is not trivial. Documents parts are not meant to be standalone units of content, and disambiguation requires some level of cross-referencing. The biggest performance gain on my retrieval evaluation in past projects (especially within the French public services) was the systematic inclusion of title hierarchy (document > chapter > section, etc.) - in long documents general information on the topic is given at this scale, but not recalled in specific paragraphs.
+
+Proper contextual support for original data is not only a concern for retrieval but also for generation. Even powerful LLMs are heavily sensitive to document structure. Lack of proper segmentation or unusual structure will result in poor tokenization (typically, merging a document separator with a token) and a higher rate of omissions and hallucinations. With tables, a hacky (but verbose) fix I’ve found is to recall the row name/column name couple before each entry.
+
```

Comment:
@morkapronczay "Even powerful LLMs" - do you think the intented menaning is "LLMs that are powerful (as opposed to less powerful LLMs)..." or "LLMs, (all of) which are powerful..."?

Reply:
Rather the first one. May be "frontier LLM" will get the meaning better.